source/StarField.cpp

```diff
@@ -90,57 +90,61 @@ void StarField::SetHaze(const Sprite *sprite)
 
 void StarField::Draw(const Point &pos, const Point &vel, double zoom) const
 {
-	glUseProgram(shader.Object());
-	glBindVertexArray(vao);
+	// Draw the starfield unless it is disabled in the preferences.
+	if(Preferences::Has("Draw starfield"))
```

Comment:
I think it would be better to write the opposite, so that you avoid adding an indentation level to the whole function and it reduces the git change. More readable too. It would be a good idea to cut this function in two though

```c++
if(not Preference::Has("Draw starfield"))
    return;
```

Reply:
I thought the same thing, but the reason it's done this way is so that having the "Draw starfield" preference off doesn't also skip the "Draw background haze" section lower down in this function.

Reply:
So why not cutting this function in two ? Because the second part is not dependent of the first part, it does not make sens to keep them together. Maybe just add a `drawBackgroundHaze`, and call it at the end of this function